examples/augmented_inference.jl

```diff
@@ -0,0 +1,82 @@
+using AbstractGPs
+using TemporalGPs
+using Distributions
+using StatsFuns
+using LinearAlgebra
+using ProgressMeter
```

Comment:
Could we maybe remove the ProgressMeter stuff? Just adds an extra dep and it's not totally necessary.